src/tests/ftest/dfuse/daos_build.py

```diff
@@ -280,3 +281,42 @@ def run_build_test(self, cache_mode, il_lib=None, run_on_vms=False):
                 self.fail(f'{fail_type} over dfuse in mode {cache_mode}')
 
         self.log.info('Test passed')
+
+    def _get_daos_build_checkout(self):
+        """Try to get the SHA or branch to checkout for the version of daos being tested.
+
+        Returns:
+            str: The SHA or branch name to checkout
+        """
+        # Try to get the SHA direct from daos version -j
+        daos_version_response = None
+        try:
+            daos_version_response = self.get_daos_command().version()["response"]
+            return daos_version_response["revision"]
+        except KeyError:
+            pass
+
+        # Try to get the SHA from from the RPM name
+        if self.prefix.startswith("/usr"):
+            result = run_remote(
+                self.log, NodeSet(self.hostlist_clients[0]), "rpm -qf $(which daos)")
+            if result.passed:
+                package = list(result.all_stdout.values())[0]
+                search = re.findall(r"\.g([0-9a-z]+)", package)
+                if search:
+                    return search[0]
+
+        # Try to determine the version from daos version -j
+        if daos_version_response:
+            try:
+                major, minor = daos_version_response["version"].split(".")[:2]
+                if bool(int(minor) % 2):
+                    # Assume odd minor version corresponds to master
+                    return "origin/master"
+                # Assume even versions correspond to a release branch
+                return f"origin/release/{major}.{minor}"
```

Comment:
AFAIK, this part can't be tested until some RC or TB runs the test, or we generate some special RPMs to test this

Reply:
We could manually install a release candidate RPM and then copy this file on top of the install to verify lines 309-317